src/screens/staking/lib/staking_sdk/wallet_operations/cosmos.ts

```diff
@@ -118,6 +129,136 @@ const getCosmosError = (err: Error): CosmosError => {
   );
 };
 
+type PollTxOpts = {
+  rpc: string;
+  txHash: string;
+};
+
+const pollTx = async ({ rpc, txHash }: PollTxOpts) => {
+  const tmClient = await connectComet(rpc);
+
+  const queryClient = QueryClient.withExtensions(tmClient, setupTxExtension);
+
+  while (true) {
+    try {
+      const tx = await queryClient.tx.getTx(txHash);
+
+      return tx;
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.log("debug: cosmos.ts: err", err);
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 5000));
+  }
+};
+
+type AccountInfo = {
+  account_number: string;
+  sequence: number;
+};
+
+const signAndBroadcastEthermint = async (
+  rpc: string,
+  account: Account,
+  latestAccountInfo: AccountInfo,
+  client: SigningStargateClient,
+  messages: EncodeObject[],
+  fee: StdFee,
+  memo: string,
+) => {
+  const signer = (account.wallet === WalletId.Leap
+    ? window.leap!.getOfflineSigner(account.networkId)
+    : window.keplr!.getOfflineSigner(
+        account.networkId,
+      )) as unknown as OfflineDirectSigner;
+
+  const { account_number: accountNumber, sequence } = latestAccountInfo;
+
+  if (typeof sequence !== "number" || !accountNumber) {
```

Comment:
Typescript infers `number` for the `sequence` variable. Do you think we could get rid of the type check?

Reply:
The thing is these values are required for the transaction to work, so was adding a check to make sure they are present. Initially it was using `!sequence` but the sequence here can be `0`. And I wouldn't try this `sequence > -1` since that is `true` for `null`.

Reply:
In that case if Typescript is not lying you could get rid of the check altogether. Since `sequence` doesn't have `null` or `undefined`.

Reply:
Because these values are not local but from an API response, I think we can't completely rely on them. Alternative would be to update the types but since they are synced with the backend, that would allow not passing them from there. I personally like to add these type of checks for API responses as some sort of defensive programming, either with a library like zod or validator or manually like here (since the types exist, I didn't want to create a schema for this).

Reply:
I understand better now, I think this is less than ideal since we are trading safety, DX, and time for convenience.

You have the discipline to work like that but I might not even see the issue myself or someone else.

I think we should also work on our approach when it comes to those issues. I mean we control both parts of the code. We should be able to make our lives easier.

Anyway, I don't think this is the right place to address this issue and I don't want to hinder any progress.